### PR TITLE
fix: fixed test cases and empty cluster name

### PR DESCRIPTION
**Pull request by voorepreethi**
#### What this PR does / why we need it:
This Pull Request addresses a bug where the cluster name was not included in the values file, resulting in an inability to append an empty cluster name when adding labels. This fix ensures that an empty cluster name is appended by default, thereby allowing labels to be correctly appended in the Prometheus scraper configuration.

Details
Issue: The absence of a cluster name in the values file prevented the addition of an empty value for the cluster name during the label configuration in Prometheus scrapers.
Fix: The values file has been updated to include an empty cluster name by default. This update facilitates the proper addition of labels, even when no specific cluster name is provided.
Impact: Ensures that Prometheus scrapers can append labels successfully, thereby maintaining consistent and accurate monitoring metrics.
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml [hunk 1]
```diff
@@ -5,10 +5,38 @@ release:
   name: my-release
   namespace: my-namespace
 tests:
-  - it: "additional labels for entity synthesis are included by default"
-    templates:
-      - templates/configmap.yaml
+  - it: "includes additional labels with cluster name empty"
```
Comment (rohit-bandlamudi-nr) [inline charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml:8]: nit: Update test cases names to include info about cluster value and the resulting cluster_name

1. cluster_name is "" when cluster is not set
2. cluster_name is "" when cluster is set with empty value
3. cluster_name is "" when cluster is set with null value
4. cluster_name is "" when cluster is set with empty string
5. cluster_name is set properly when cluster is set with non-empty

**Review by rohit-bandlamudi-nr — CHANGES_REQUESTED** [review]

File: charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml (hunk 1 above)
Comment (rohit-bandlamudi-nr) [reply charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml:8]: Also include another case for quoted cluster name. for eg. cluster: "test"

Comment (maya-jha) [reply charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml:8]: Maybe instead of saying 
`cluster_name is "" when cluster is not set` we  can say its empty `cluster_name is empty when cluster is not set`

File: charts/newrelic-logging/templates/_helpers.tpl [hunk 2]
```diff
@@ -205,7 +205,12 @@ Returns fluentbit config to collect and forward its metrics to New Relic
     add_label            source kubernetes
     add_label            pod_name ${HOSTNAME}
     add_label            node_name ${NODE_NAME}
-    {{- printf "add_label            cluster_name %s" (include "newrelic-logging.cluster" .) | nindent 4 -}}
+    {{- $clusterName := (include "newrelic-logging.cluster" .) -}}
```
Comment (maya-jha) [inline charts/newrelic-logging/templates/_helpers.tpl:208]: Where does it get deployed if we don't provide cluster name? Does it take it from kube config? How about other params , can they be null too?

File: charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml [hunk 3]
```diff
@@ -5,10 +5,38 @@ release:
...
+    set:
+      cluster: ""
```
Comment (maya-jha) [inline charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml:10]: should we test based on `[newrelic-logging.cluster](https://github.com/newrelic/helm-charts/blob/85037884f7e3c5a975a47063ab724e20f9c4e124/charts/newrelic-logging/templates/_helpers.tpl#L63)` instead of cluster? Also, I think we may need to just test based on empty. We can add a test for `newrelic-logging.cluster` itself if it does not exist.

File: charts/newrelic-logging/templates/_helpers.tpl [hunk 4]
```diff
@@ -205,7 +205,12 @@ Returns fluentbit config to collect and forward its metrics to New Relic
...
+    {{- if $clusterName -}}
+    {{- printf "add_label            cluster_name %s" $clusterName | nindent 4 -}}
+    {{- else -}}
+    {{- printf "add_label            cluster_name \"%s\"" $clusterName | nindent 4 -}}
+    {{- end -}}
     {{- printf "add_label            namespace %s" .Release.Namespace | nindent 4 -}}
```
Comment (maya-jha) [inline charts/newrelic-logging/templates/_helpers.tpl:214]: can any other label be also null/empty?

**Review by maya-jha — CHANGES_REQUESTED** [review]

File: charts/newrelic-logging/templates/_helpers.tpl (hunk 4 above)
Comment (voorepreethi) [reply charts/newrelic-logging/templates/_helpers.tpl:214]: charts won't get deployed without release name . So other parameters are mandatory . Cluster name is something which we are fetching not from the deployed cluster but from values

File: charts/newrelic-logging/templates/_helpers.tpl (hunk 2 above)
Comment (rohit-bandlamudi-nr) [reply charts/newrelic-logging/templates/_helpers.tpl:208]: other params, cant be null as they will fetched from the generated k8s manifests. Cluster name would be picked up from values file

File: charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml (hunk 3 above)
Comment (rohit-bandlamudi-nr) [reply charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml:10]: The value `newrelic-logging.cluster` is defined by us in the helpers.tpl file and its value depends on the value given in values file. This test is sufficient as we can know the resultant config based on value  of cluster given in values file. Separate test for the value of newrelic-logging.cluster can be added but that's not added till now and we tested it manually anyways.

File: charts/newrelic-logging/templates/_helpers.tpl (hunk 2 above)
Comment (rohit-bandlamudi-nr) [reply charts/newrelic-logging/templates/_helpers.tpl:208]: > Where does it get deployed if we don't provide cluster name? 

Value of cluster name is only a label we could use later for entity synthesis and relationship synthesis. Doesn't affect location of deployment. If that is empty, it would not be easy to query the fluent bit entity based on the empty cluster name

File: charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml (hunk 3 above)
Comment (maya-jha) [reply charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml:10]: Thats not a good idea. We should test based on `newrelic-logging.cluster` and then also add a test for `newrelic-logging.cluster`. Otherwise, we will need to add a test here based on both local `cluster` and `global.cluster`

File: charts/newrelic-logging/templates/_helpers.tpl (hunk 4 above)
Comment (maya-jha) [reply charts/newrelic-logging/templates/_helpers.tpl:214]: can we get cluster name from the deployed cluster?

File: charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml (hunk 3 above)
Comment (voorepreethi) [reply charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml:10]: > then also add a test for newrelic-logging.cluster

we can have a separate test for this to see if newrelic-logging.cluster is properly set (whether global or local one is chosen). But leave the existing test (based on `cluster` from values file) as it is, as we want to check if  cluster_name is properly set based on the value of "cluster" from values file

Comment (rohit-bandlamudi-nr) [reply charts/newrelic-logging/tests/fluentbit_monitoring_config_test.yaml:10]: @voorepreethi see if its possible to test with newrelic-logging.cluster variable.

File: charts/newrelic-logging/templates/_helpers.tpl (hunk 4 above)
Comment (rohit-bandlamudi-nr) [reply charts/newrelic-logging/templates/_helpers.tpl:214]: AFAIK, it's not possible. We have to check. But the idea here is to not restrict the cluster to take the value of the cluster name that the release is deployed on. The user can give any value he wants here. This depends on what is the name of the cluster entity that the user want to create / associate related entities to.

Comment (maya-jha) [reply charts/newrelic-logging/templates/_helpers.tpl:214]: create a JIRA ticket to address other comments related to test and using cluster name even if its empty.

**Review by maya-jha — APPROVED** [review]

**Review by hrai-nr — APPROVED** [review]

**Review by rohit-bandlamudi-nr — APPROVED** [review]
